File: app/frontend/src/types/escalation.ts
export type EscalationLevel = 'monitor' | 'pause' | 'owner_takeover'
export type EscalationOutcome =
  | 'approved' | 'adjusted' | 'owner_takeover' | 'release' | 'need_more_info'

export interface LeadEscalation {
  id: string
  lead_id: string
  level: EscalationLevel
  source: 'manual' | 'auto_idle'
  decision_needed: string
  summary: string
  raised_by: string | null
  raised_at: string
  status: 'open' | 'resolved'
  outcome: EscalationOutcome | null
  resolution_note: string | null
  resolved_by: string | null
  resolved_at: string | null
  lead_customer_name: string | null
  lead_status: string | null
}

export const LEVEL_LABELS: Record<EscalationLevel, string> = {
  monitor: 'Monitor',
  pause: 'Pause before quote',
  owner_takeover: 'Owner takeover',
}

export const OUTCOME_LABELS: Record<EscalationOutcome, string> = {
  approved: 'Approved',
  adjusted: 'Adjusted',
  owner_takeover: 'Owner taking over',
  release: 'Release',
  need_more_info: 'Need more info',
}

export const DECISION_OPTIONS = ['price', 'schedule', 'truck', 'release', 'owner takeover'] as const

File: app/backend/app/schemas/escalation.py
from __future__ import annotations

from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, Field

Level = Literal["monitor", "pause", "owner_takeover"]
Outcome = Literal["approved", "adjusted", "owner_takeover", "release", "need_more_info"]


class RaiseEscalationIn(BaseModel):
    level: Level
    decision_needed: str = Field(min_length=1)
    summary: str = Field(min_length=1)
    raised_by: Optional[str] = None


class ResolveEscalationIn(BaseModel):
    outcome: Outcome
    resolution_note: Optional[str] = None
    resolved_by: Optional[str] = None


class EscalationSummaryOut(BaseModel):
    summary: str


class LeadEscalationOut(BaseModel):
    id: str
    lead_id: str
    level: str
    source: str
    decision_needed: str
    summary: str
    raised_by: Optional[str] = None
    raised_at: datetime
    status: str
    outcome: Optional[str] = None
    resolution_note: Optional[str] = None
    resolved_by: Optional[str] = None
    resolved_at: Optional[datetime] = None
    # Populated via join when listing for the queue band
    lead_customer_name: Optional[str] = None
    lead_status: Optional[str] = None

    model_config = {"from_attributes": True}
